static/js/addReviewClient.js

```diff
@@ -1,35 +1,89 @@
 const addReviewButton = document.querySelector(".add-review-button");
-let addReviewContent = document.querySelector(".add-review-content");
+const addReviewContent = document.querySelector(".add-review-content");
+const loggedOutEl = document.querySelector(".add-review-logged-out");
+const loggedInEl = document.querySelector(".add-review-logged-in");
 
 addReviewButton.addEventListener("click", () => {
   addReviewButton.classList.toggle("clicked");
   addReviewContent.classList.toggle("show");
 });
 
+let loggedIn = false;
+let sendReviewToken;
+
+document.querySelector("#login").addEventListener("click", async () => {
+
+  let username = document.querySelector("#username").value;
+  let password = document.querySelector("#password").value;
+  const bufferedInput = btoa(username + ":" + password);
+  try {
+    const response = await fetch(
+    new Request("/api/token", {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        "Authentication": 
+        "Basic " + bufferedInput,
+      },
+    }));
+    if (response.status === 401) {
+      alert("Fel lösenord eller användarnamn");
+      throw new Error("Invalid username or password");
+    }
+    loggedIn = true;
+    loggedInEl.classList.add("show");
+    loggedInEl.classList.remove("hide");
+    loggedOutEl.classList.add("hide");
+
+    const data = await response.json();
+    sendReviewToken = data.token;
+
+  } catch (e) {
+    console.log(e.message);
```

Comment:
maybe remove the console.log :)

Reply:
This console log actually serves a purpose that's not debugging! If you enter the wrong password, you'll get an Error 401 printed in the console. With this, I've just added a message saying the username or password is invalid, with `throw new Error("Invalid username or password");`
But I suppose with the super fancy alert I inserted I can remove it now 😁